app/controllers/events_controller.rb

```diff
@@ -0,0 +1,75 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:show, :edit, :update, :destroy]
+  before_action :authorize_project_member
```

Comment:
I thought about using Pundit gem for authorization cause it doesnt look good when mixing this authorization logic in the controller. It's much more readable when you move this logic to some policy class.